### ISSUE TITLE
Incorrect table display on scroll

### ISSUE THREAD

**Issue by bschousek** [beeware/toga#1130]
**Describe the bug**
Scrolling a table on Windows 10 leads to incorrect display values. 

**To Reproduce**
1. Create a new Briefcase/Toga app and modify the app.py startup boilerplate as to include a table as follows:

        main_box = toga.Box()

        tbl_headings=['heading']
        tbl_data=list(range(50))
        self.table=toga.Table(headings=tbl_headings,data=tbl_data,missing_value="",style=Pack(flex=1))
        main_box.add(self.table)
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

2. run the app ('briefcase dev') to see:
![image](https://user-images.githubusercontent.com/3130008/97872140-2343e600-1cdb-11eb-9cac-ca9cea2541d5.png)
3. click the down arrow on the scrollbar twice to see corruption on the bottom list element as follows:
![image](https://user-images.githubusercontent.com/3130008/97872209-440c3b80-1cdb-11eb-83eb-f4b39374d229.png)
4. click the down arrow once more to see persistent corruption and a new wholly incorrect value:
![image](https://user-images.githubusercontent.com/3130008/97872254-5b4b2900-1cdb-11eb-8bd9-146f0b892002.png)
5. click the up arrow twice to see that now the top elements in the list are showing up as zero
![image](https://user-images.githubusercontent.com/3130008/97872300-6f8f2600-1cdb-11eb-9d1d-984ab6a29020.png)

**Expected behavior**
correct values regardless of scrolling
**Screenshots**
NA

**Environment:**
 - Operating System: Windows 10 Enterprise
 - Python version:  3.78
 - Software versions: 
   - Briefcase:0.3.3
   - Toga: 0.3.0.dev25
   

**Additional context**
The data in the table is still valid. there seems to be a connection with refresh as sometimes resizing the window fixes the problem. Also, if the window is very small it seems the problem does not present itself.

**Comment by AsessionGuy** [comment]
I found out that if you go to the end of the scroll and wide up the window, the container resets and adds blank space above the content (with approximately the same height as the content).

**Comment by t-arn** [comment]
Today, I found the exact same problem when I tried to scroll a table by clicking into the scrollbar. Then, you get a lot of seemingly  duplicated rows. When you select such a row, you'll find that the data behind is still correctly there, but is displayed incorrectly.
@freakboy3742 Do you know: has scrolling a Table on Windows ever worked correctly? What could be the reason that scrolling stopped working correctly?

**Comment by t-arn** [comment]
@bschousek I found that the problem must be in the caching somewhere.
When I disable the cache, then scrolling works correctly for me:
```
# self.native.CacheVirtualItems += self.winforms_cache_virtual_items
```
But just disabling the cache will reduce the performance of the Table :-(

**Comment by t-arn** [comment]
With PR #1238, scrolling now also works with activate cache

**Comment by freakboy3742** [comment]
This was addressed in #1238.
